src/Events/Category_Colors/Event_Category_Meta.php

```diff
@@ -75,207 +92,143 @@ public function __construct( int $term_id ) {
+	public function delete( string $key ): self {
+		$key = $this->validate_key( $key );
+
+		if ( is_wp_error( $key ) ) {
+			return $key;
 		}
 
-		/**
-		 * Filter the meta value before it is saved.
-		 *
-		 * @since TBD
-		 *
-		 * @param mixed $value   The sanitized meta value.
-		 * @param int   $term_id The term ID the meta value belongs to.
-		 */
-		return apply_filters( 'tec_events_category_validate_meta_value', $value, $this->term_id );
+		$this->pending_deletes[] = $key;
+
+		return $this;
 	}
 
 	/**
-	 * Gets metadata for the term.
+	 * Save all queued meta updates and deletions.
 	 *
 	 * @since TBD
 	 *
-	 * @param string|null $key Optional. The meta key to retrieve.
-	 *
-	 * @return mixed The meta value, or WP_Error if invalid.
+	 * @return $this
 	 */
-	public function get( ?string $key = null ) {
-		if ( null === $key ) {
-			$all_meta = $this->get_meta();
-
-			return array_map(
-				function ( $values ) {
-					return is_array( $values )
-						&& isset( $values[0] ) ? $values[0] : $values;
-				},
-				$all_meta
-			);
+	public function save(): self {
+		foreach ( $this->pending_updates as $key => $value ) {
+			update_term_meta( $this->term_id, $key, $value );
 		}
 
-		$key = $this->validate_key( $key );
-		if ( is_wp_error( $key ) ) {
-			return $key;
+		foreach ( $this->pending_deletes as $key ) {
+			delete_term_meta( $this->term_id, $key );
 		}
 
-		$value = $this->get_meta( $key );
+		// Clear queues after saving.
+		$this->pending_updates = [];
+		$this->pending_deletes = [];
 
-		return ( '' !== $value ) ? $value : null;
+		return $this;
 	}
 
+
 	/**
-	 * Sets metadata for the term.
+	 * Retrieves metadata for the term.
 	 *
 	 * @since TBD
 	 *
-	 * @param string $key   The meta key to update.
-	 * @param mixed  $value The value to store.
+	 * @param string|null $key Optional. The meta key to retrieve.
 	 *
-	 * @return $this|WP_Error Returns the instance for chaining or WP_Error if invalid.
+	 * @return mixed The meta value, or an array of all metadata if no key is provided.
 	 */
-	public function set( string $key, $value ) {
-		$key   = $this->validate_key( $key );
-		$value = $this->validate_value( $value );
-
-		if ( is_wp_error( $key ) || is_wp_error( $value ) ) {
-			return is_wp_error( $key ) ? $key : $value;
+	public function get( ?string $key = null ) {
+		if ( null === $key ) {
+			return get_term_meta( $this->term_id );
 		}
 
-		update_term_meta( $this->term_id, $key, $value );
-
-		/**
-		 * Fires after metadata has been updated.
-		 *
-		 * @since TBD
-		 *
-		 * @param int    $term_id The term ID where metadata was set.
-		 * @param string $key     The meta key that was updated.
-		 * @param mixed  $value   The new value that was stored.
-		 */
-		do_action( 'tec_events_category_set_meta', $this->term_id, $key, $value );
+		$key   = $this->validate_key( $key );
+		$value = get_term_meta( $this->term_id, $key, true );
 
-		return $this;
+		return ( '' !== $value ) ? $value : null;
```

Comment:
I think this would work better (in case it returns false)
```suggestion
		empty( $value ) ? null : $value;
```

Reply:
I'm not sure if `empty()` is the best way to go. Empty will return true if the value is `0` as well.

Reply:
I ended up creating a new test `it_should_properly_handle_empty_values_when_retrieving_meta` to test different scenarios. `empty()` didn't work. However, I ended up refactoring the logic to better handle the different scenarios.